build.gradle

```diff
@@ -309,25 +304,10 @@ subprojects {
     options.compilerArgs << "-Xlint:-serial"
     options.compilerArgs << "-Xlint:-try"
     options.compilerArgs << "-Werror"
-
-    // --release is the recommended way to select the target release, but it's only supported in Java 9 so we also
-    // set --source and --target via `sourceCompatibility` and `targetCompatibility` a couple of lines below
-    if (JavaVersion.current().isJava9Compatible())
-      options.release = minJavaVersion
-    // --source/--target 8 is deprecated in Java 20, suppress warning until Java 8 support is dropped in Kafka 4.0
-    if (JavaVersion.current().isCompatibleWith(JavaVersion.VERSION_20))
-      options.compilerArgs << "-Xlint:-options"
-
+    options.release = minJavaVersion
     addParametersForTests(name, options)
   }
 
-  java {
-    // We should only set this if Java version is < 9 (--release is recommended for >= 9), but the Scala plugin for IntelliJ sets
-    // `-target` incorrectly if this is unset
-    sourceCompatibility = minJavaVersion
-    targetCompatibility = minJavaVersion
-  }
```

Comment:
Is this correct? Does the Scala plugin no more suffer from the issue described in the comment? 
KIP-750 to modify these not to delete them.

Reply:
Thanks for pointing out.
I think it is a mistake, I will add it back

Reply:
I've checked the latest Scala plugin, and the `-target` option is no longer included.
```
        <parameter value="-release" />
        <parameter value="11" />
```
Additionally, based on the discussion (https://youtrack.jetbrains.com/issue/SCL-21039/Dont-set-parameter-value-target-in-.idea-scalacompiler.xml-when-Gradle-build-is-using-release-flag-applies-for-Gradle-8.0.1#focus=Comments-27-7380486.0-0), it seems the issue has already been fixed.

Reply:
Thanks for information 😺

Reply:
@frankvicky Could you please also update KIP-750?

Reply:
done

Reply:
@chia7712 @frankvicky looks like this removal of sourceCompatibility, targetCompatibility Java compile options from build.gradle is not setting the Scala compile target version correctly. I have prepared a build with JDK17 and getting UnsupportedClassVersionError for Scala classes while trying to start with JDK 11.

Reply:
I tried to fix here: https://github.com/apache/kafka/pull/17546